service_implementation/advanced_plm.md

```diff
@@ -1 +1,456 @@
+- REST end point: https://caxman.clesgo.net/jotne/GSS/resources/plm
+
+GSS covers a small subset of simplified methods (services) of the PLM Server. The remaining PLM Server functionality is only available as native PLM Server SOAP web-services; see below for details.
+
+Java (Maven) client based examples can be retrieved from here: https://github.com/CAxMan/PlmGSS-samples
+
+The PLM Server is also accessible via the GSS.
+Code samples of that are here: https://github.com/CAxMan/infrastructureClients
+
+The following GSS methods (SOAP) are only implemented for the PLM Server storage option, not for the SWIFT storage:
+- *createFolderVersion*
+- *listFolderVersions*
+- *listFileVersions*
+- *plm2gss*
+- *gss2plm*
+
+**createFolderVersion** - creates a new version of the specified folder. Children folders of the existing selected version are reused; files are copied.
+Parameters:
+- **folderID** - GSS folder id of the folder to create the new version from. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object with GSS folder ID of the newly created folder version
+
+**listFolderVersions** - lists all versions of the specified folder including the one specified.
+Parameters:
+- **path** - GSS folder id. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object for each folder version
+
+**listFileVersions** - lists all versions of the specified file including the one specified.
+Parameters:
+- **path** - GSS file id. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object for each file version
+
+The following two methods can be used to switch between GSS and PLM native services.
+GSS uses object identifiers like the following.
+"plm://TestRepository/MyModel123/1234567890"
+where the last part (integer value) is a PLM object identifier.
+
+**plm2gss** - converts a PLM object identifier to a GSS identifier
+
+Example:
+```Java
+String keystoneToken = ...
+V_node node = ...
+ResourceInformation folderId = srv.plm2Gss(node.getNodus().getItem().getInstance_id(), keystoneToken);
+String id = folderId.getUniqueName();
+```
+
+**gss2plm** - converts a GSS identifier to PLM object identifier
+
+Example:
+```Java
+String keystoneToken = ...
+FileWebService_PortType srv = new FileWebService_ServiceLocator().getFileWebServicePort();
+PlmResource plmId = srv.gss2Plm("plm://TestRepository/MyModel123/1234567890", keystoneToken);
+
+String wsdl = plmId.getWsdlUrl();
+String serviceUrl = plmId.getServiceUrl();
+```
+
+## PLM API through native services (SOAP)
+
+The remaining and largest share of PLM functionality, that is, for which GSS does not deliver endpoints, is only available for use in CAxMan through PLM Server native services.

```

Comment:
Suggested addition at the end of this line:
Note that in order to use this functionality, GSS URIs need to be converted into native PLM object identifiers (see above for converter functions).

Reply:
done